xcm/pallet-xcm/src/migration.rs

```diff
@@ -25,6 +25,7 @@ const DEFAULT_PROOF_SIZE: u64 = 64 * 1024;
 
 pub mod v1 {
 	use super::*;
+	use crate::{CurrentMigration, VersionMigrationStage};
 
 	pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
 	impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
```

Comment:
We should remove the `pre_upgrade` hook here, otherwise on-runtime-upgrade will start breaking after this upgrade is enacted on-chain

Reply:
Could you do it now please, or better yet use a `VersionedRuntimeUpgrade` which was designed specifically for these use cases? 

We do not want runtime upgrade hooks to pass or fail depending on whether or not an upgrade has run. The approach of "Remove it next time you run an upgrade" is flakey and will cause our CI to break in the period between the upgrade and removing it.

Reply:
so how is that being tested in CI? against which runtime version?

Reply:
it's tested using `try-runtime-cli` using the current on-chain state. 

see jobs `gitlab-check-runtime-migration-polkadot, `gitlab-check-runtime-migration-westend`, etc